pkg/controller/plan/controller.go

```diff
@@ -275,14 +275,32 @@ func (r *Reconciler) updatePlanStatus(plan *api.Plan) error {
 	// Record events.
 	r.Record(plan, plan.Status.Conditions)
 
-	// Apply changes.
-	plan.Status.ObservedGeneration = plan.Generation
+	// Try to fetch latest version first to get current ResourceVersion
+	latest := &api.Plan{}
+	key := client.ObjectKeyFromObject(plan)
+	if err := r.Get(context.TODO(), key, latest); err != nil {
+		r.Log.V(1).Info("Failed to fetch latest plan version, trying update with current version",
+			"plan", plan.Name, "error", err.Error())
+
+		// Fallback: try update with current plan
+		plan.Status.ObservedGeneration = plan.Generation
+		if updateErr := r.Status().Update(context.TODO(), plan.DeepCopy()); updateErr != nil {
+			r.Log.Error(updateErr, "Failed to update plan status with current version")
+			return updateErr
+		}
+		return nil
+	}
+
```

Comment:
@yaacov ptal this could be interesting for you as you were looking in the conflits